python/cudf/cudf/testing/testing.py

```diff
@@ -201,31 +206,50 @@ def assert_column_equal(
     ):
         left = left.astype(left.categories.dtype)
         right = right.astype(right.categories.dtype)
-
     columns_equal = False
-    try:
-        columns_equal = (
-            (
-                cp.all(left.isnull().values == right.isnull().values)
-                and cp.allclose(
-                    left[left.isnull().unary_operator("not")].values,
-                    right[right.isnull().unary_operator("not")].values,
+    if left.size == right.size == 0:
+        columns_equal = True
+    elif not (
+        (is_string_dtype(left) and is_numeric_dtype(right))
+        or (is_numeric_dtype(left) and is_string_dtype(right))
+    ):
+        try:
+            columns_equal = (
+                (
+                    (
+                        cp.all(left.isnull().values == right.isnull().values)
+                        if not (left.dtype.kind == right.dtype.kind == "f")
+                        else (
+                            cp.all(is_nan(left).values == is_nan(right).values)
+                            and cp.all(
+                                left.isnull().values == right.isnull().values
+                            )
+                        )
+                    )
+                    and cp.allclose(
+                        left[left.isnull().unary_operator("not")].values,
+                        right[right.isnull().unary_operator("not")].values,
+                    )
                 )
+                if not check_exact and is_numeric_dtype(left)
+                else left.equals(right)
             )
-            if not check_exact and is_numeric_dtype(left)
-            else left.equals(right)
-        )
-    except TypeError as e:
-        if str(e) != "Categoricals can only compare with the same type":
-            raise e
-        if is_categorical_dtype(left) and is_categorical_dtype(right):
-            left = left.astype(left.categories.dtype)
-            right = right.astype(right.categories.dtype)
+        except TypeError as e:
+            if str(e) != "Categoricals can only compare with the same type":
+                raise e
+            if is_categorical_dtype(left) and is_categorical_dtype(right):
+                left = left.astype(left.categories.dtype)
+                right = right.astype(right.categories.dtype)
     if not columns_equal:
-        msg1 = f"{left.values_host}"
-        msg2 = f"{right.values_host}"
+        ldata = [val for val in left.to_pandas(nullable=True)]
```

Comment:
Side question - I've wondered how reliable `nullable=True` is given pandas support for nullable float dtype is still non-public?

Reply:
It seems like they're being fairly forward with these [as of 1.2.0](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.2.0.html?highlight=float64dtype#experimental-nullable-data-types-for-float-data)